rasa/core/training/story_reader/yaml_story_reader.py

```diff
@@ -79,118 +100,147 @@ def read_from_parsed_yaml(
         ):
             return []
 
-        stories = parsed_content.get(KEY_STORIES, [])
-        self._parse_data(stories, is_rule_data=False)
+        for key, parser_class in {
+            KEY_STORIES: StoryParser,
+            KEY_RULES: RuleParser,
+        }.items():
+            stories = parsed_content.get(key, [])
+            parser = parser_class.from_reader(self)
+            parser.parse_data(stories)
+            self.story_steps.extend(parser.get_steps())
+        return self.story_steps
 
-        rules = parsed_content.get(KEY_RULES, [])
-        self._parse_data(rules, is_rule_data=True)
+    @staticmethod
+    def is_yaml_story_file(file_path: Text) -> bool:
```

Comment:
it might be easier to review commit by commit as I did a refactoring in the class